### PR TITLE
WIP - Backoffice alerts view

### PR DESCRIPTION
**Pull request by Omri-Levy**
- feat: added initial markup for transaction monitoring alerts
- refactor(backoffice-v2): made the alerts view look closer to Figma design
- feat: alerts
- feat: alerts
- feat: pagination
- feat: faker
- feat: faker alerts
- feat: apply filters
- fix: swagger
- fix: swagger
- refactor(backoffice-v2): addressed sign off comments
- feat(backoffice-v2): added fetchers and schemas for alerts
- feat: added alerts analysis sheet & table
- feat(backoffice-v2): added fallback states for the alerts table
- feat(backoffice-v2): added opening sheet on row click
- feat(backoffice-v2): added sorting
- fix(blocks-package): omri-levy/fix/blocks-types
- feat(backoffice-v2): added pagination for the alerts table
- feat(backoffice-v2): implemented search client-side (not api yet)
- feat(backoffice-v2): added multi select filtering

### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: apps/backoffice-v2/src/main.tsx [hunk 1]
```diff
@@ -3,15 +3,14 @@ import { createRoot } from 'react-dom/client';
...
-import { Router } from './Router/Router';
 import { env } from './common/env/env';
 import './i18n';
 import './index.css';
+import { Router } from '@/Router/Router';
 
 const rootElement = document.getElementById('root');
 
-// Avoid race conditions when using the mock server.
+/// Avoid race conditions when using the mock server.
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/main.tsx]: Very important.

File: apps/backoffice-v2/src/pages/TransactionMonitoring/TransactionMonitoring.tsx [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import React, { FunctionComponent } from 'react';
+import { Outlet } from 'react-router-dom';
+
+export const TransactionMonitoring: FunctionComponent = () => {
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/TransactionMonitoring/TransactionMonitoring.tsx:4]: Is this really needed?

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertAnalysisSheet/AlertAnalysisSheet.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+    <Sheet open={isOpen} onOpenChange={onOpenStateChange}>
+      <SheetContent side="bottom" className={`w-full`}>
+        <div className="flex flex-col p-4">
+          <div className="flex flex-col gap-4">
+            <h2 className="text-2xl font-bold">High Velocity - Inbound - Counterparty</h2>
+            <div className="mb-10 flex flex-col gap-1">
+              <h3 className="text-sm font-bold">Summary</h3>
+              <p className="max-w-[88ch] text-sm text-[#64748B]">
+                The Customer has received 5 inbound transactions from the same counterparty within 1
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertAnalysisSheet/AlertAnalysisSheet.tsx:25]: I don't think was supposed to be hard-coded...

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertAnalysisSheet/components/AlertAnalysisTable/columns.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+          <span className={`font-semibold`}>{date}</span>
+          <span className={`text-xs text-[#999999]`}>{time}</span>
+        </div>
+      );
+    },
+    header: 'Date & Time',
+  }),
+  columnHelper.accessor('transactionId', {
+    cell: info => {
+      const value = info.getValue();
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertAnalysisSheet/components/AlertAnalysisTable/columns.tsx:36]: The inner functions repeat themselves - one with `text-sm` className and one with `font-semibold` as well. It looks like they could be extracted to functions and re-used

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+const columnHelper = createColumnHelper<TAlertsList[number]>();
+
+export const columns = [
+  columnHelper.accessor('dataTimestamp', {
+    cell: info => {
+      const value = info.getValue();
+
+      if (!value) {
+        return <TextWithNAFallback>{value}</TextWithNAFallback>;
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx:24]: If `value` is falsy, what are we gonna really see?

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/get-alerts-search-schema.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      >)
+      .catch('dataTimestamp'),
+    filter: z
+      .object({
+        assigneeId: z.array(z.string().nullable()).catch([authenticatedUserId, null]),
+        status: z.array(z.enum(AlertStatuses)).catch([AlertStatus.NEW]),
+        state: z.array(z.string().nullable()).catch([]),
+      })
+      .catch({
+        // assigneeId: [authenticatedUserId, null],
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/get-alerts-search-schema.ts:20]: Why is it commented out?

**Review by tomer-shvadron — APPROVED** [review]

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertAnalysisSheet/AlertAnalysisSheet.tsx (hunk 3 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertAnalysisSheet/AlertAnalysisSheet.tsx:25]: This code is supposed to be as part of the transactions drawer task that was unintentionally merged into my branch. Work will resume when I start working on the drawer task.

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx (hunk 5 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx:24]: `N/A` with styling.

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/get-alerts-search-schema.ts (hunk 6 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/get-alerts-search-schema.ts:20]: There are still integration pains. Currently fields that shouldn't be required are required.

**Pull request merged**